apps/editing-toolkit/editing-toolkit-plugin/tutorials/class-wpcom-tutorials.php

```diff
@@ -0,0 +1,246 @@
+<?php
+/**
+ * Tutorials (We would have called them Guided Tours but it's been done).
+ *
+ * Each Tutorial contains a flat list of Tasks. Each Task may contain several Steps in
+ * a `WpcomTourKit` component on the frontend, but we are only concerned with the top
+ * two levels of Tutorial and its contained Tasks, tracking Task completion status
+ * on the backend. This API will be used to generate
+ *
+ * @package A8C\FSE
+ */
+
+/**
+ * WPCom_Tutorials class
+ */
+class WPCom_Tutorials {
+
+	/**
+	 * Storage for our singleton instance.
+	 *
+	 * @var null|WPCom_Tutorials
+	 */
+	private static $instance = null;
+
+	/**
+	 * Internal storage for registered Tutorials.
+	 *
+	 * @var array
+	 */
+	private $registry = array();
+
+	/**
+	 * Gets aour singleton instance, initiating it the first time.
+	 *
+	 * @return WPCom_Tutorials instance.
+	 */
+	public static function get_instance() {
```

Comment:
I'm wary of singleton usage in this case since I'm uncertain if we absolutely need to guarantee only one single instance is ever created, and the downsides for unit testing are significant.  I recognize that this is a pattern used throughout WordPress though, so wondering if you have more thoughts @mattwiebe.

Reply:
Oh my mistake. I thought [the code examples](https://github.com/WordPress/wordpress-develop/blob/8277055e37c19cd69cbf829478004b3445c8bb79/src/wp-includes/functions.wp-scripts.php#L20-L28) you linked in your previous comment were already implemented in this PR. I'm definitely on board with porting over that logic into this PR to allow us to reset registry state